### DISCUSSION TITLE
Transcoding - is the clients-ID fixed?

### DISCUSSION THREAD

**Discussion by udippel** [gerbera/gerbera#3055] [General]
Sometimes laziness turns out to be useful.
I tried some more adaptation on my ancient clients, with (like in the example):

```
<transcoding enabled="yes">
  <mimetype-profile-mappings>
    <transcode mimetype="audio/x-flac" client-flags="TRANSCODING1" using="audio2mp3" />
   </mimetype-profile-mappings>
  <profiles>
   [...]
   </profiles>
</transcoding>
<clients enabled="yes">
  <client ip="192.168.120.42" flags="TRANSCODING1"></client>
</clients>

```
Doing the job perfectly okay. VERY different from the example, perfectly okay. 
Since I don't like identifiers like TRANSCODING1, subsequently I changed it to something more descriptive here. Alas, then it failed. It just didn't do that specific transcoding. I moved back to TRANSCODING1, and now it works, once again.

I don't think that I made a typo. The old person that I am, I did a :%s/Var1/Var2/g in vi, forth and back. 

Now, before exploring where actually it fails, I am lazy enough to ask, if this can be explained; or maybe is intended; and one has to use TRANSCODE1 to make it work?

The only thing I could imagine: the modified identifiers contained underscores, like "WMV_2_TV3". Since it IS a lengthy path, close down, edit, restart, pick up some WMV on that client ... I think it ought to be permissible to ask here first.

**Comment by whyman** [comment]
What you defining are client flags, they are not strings they are "bits" in a mask that we have given friendly names 

> If the flags match the ones defined in clients, the profile is selected for that client. There are are TRANSCODE1, TRANSCODE2, TRANSCODE3 or choose an unused flag, e.g. 0x10000, to avoid collisions with other features.

Transcoding Profiles: https://docs.gerbera.io/en/stable/config-transcode.html#profiles
Device Flags: https://docs.gerbera.io/en/stable/supported-devices.html#device-flags

**Reply by udippel** [reply]
Okay, thanks for the explanation. It ought to be even more clearly pointed out in the documentation, though, since the naming of e.g. the profiles may be any_friendly_name; and I have defined a large bunch, describing what they do, and why. 
Only your first cited source has the content as cited, and somewhat further down; without stating that their naming ("TRANSCODING") are not examples, but requirements, somewhat unusual in my opinion. 

I do not doubt that there is a good reason not to allow descriptive names here, though until now I wasn't aware of any. 

And thanks for the reply, saving me actually plenty of time of editing, starting, shutting down, scrolling to specific resources on specific client boxes; to find out in the end - if I understand you correctly - that any client-based transcoding has to be "TRANSCODINGN" for a client to be specified.

**Reply by KarlStraussberger** [reply]
You can use any of the friendly flag names if you don't have this kind of device, but it enables client functionality.

If you are running out of `TRANSCODINGn` you should use flags as numbers or hex numbers that are above the existing range. Problem: in the future they can become "active" client flags.

**Reply by udippel** [reply]
Thanks, @KarlStraussberger, I think I understood that. 
Just the curiosity that seems to fail at killing cats: Why do client flags have to be bit masks combined with a YELLING string, instead of just_any_string that doesn't coincide/conflict with another, predefined just_any_string? 

Or, asking the other way round, while the profiles are checked thoroughly at startup, and some unused one is pointed out, why does the check - in my case did the check - not find missing correspondences in client flags compared to transcoding profiles?

**Reply by KarlStraussberger** [reply]
As a matter of fact, clients flags must be combinable (SAMSUNG & WIERD) to get special treatment in responses. So your client will in addition require some transcoding.

Obviously the parser for the string and bits is ignoring unknown stuff. Doesn't really kill the cat.

**Reply by udippel** [reply]
The cat remains curious. Though we may well close this, because it can be made to work, despite YELLING being so ugly, last century, and the nomenclatura reminds one of DDOS and 8.3.

Oh well, a clearer hint in the documentation could still be useful.
